TexasInstruments/CD4017B/GHIElectronics.TinyCLR.Drivers.TexasInstruments.CD4017B/CD4017B.cs

```diff
@@ -0,0 +1,32 @@
+﻿using GHIElectronics.TinyCLR.Devices.Gpio;
+
+namespace GHIElectronics.TinyCLR.Drivers.TexasInstruments.CD4017B {
+    public class CD4017B {
+        private readonly GpioPin clock;
+        private readonly GpioPin reset;
+        private int currentCount;
+
+        public CD4017B(int clockPin, int resetPin) {
+            var gpio = GpioController.GetDefault();
```

Comment:
Follow the pattern of the other drivers where the GpioPins are passed in. See https://github.com/ghi-electronics/TinyCLR-Drivers/blob/dev/ShijiLighting/APA102C/APA102C.cs

Reply:
Because the user may want to use a different GPIO controller, not the built in one. Like if it's indirected over a GPIO chip. Every device interface for a driver should always be passed in as a GpioPin, SpiDevice, etc. Never an ID.

Reply:
Done!